internetarchive/openlibrary
# Removed sort parameter from canonical link

Closes #2864 


Hotfix

### Technical

Wasn't sure what other potential query parameters there could be, so I made it as general as possible so as not to break other canonical URLs.

If the only query parameter is `?sort=new` it will remove the entire query, if there are other query parameters included such as `?sort=new&some_other=query` then query will become `?some_other=query`, and the same is true if ordered like `?some_other=query&sort=new`.

According to [this StackOverflow response](https://webmasters.stackexchange.com/questions/114622/should-query-strings-be-included-or-removed-from-the-canonical-tag) an argument could be remade to remove query from the canonical URL entirely, but I wasn't sure if there were other query parameters that specify page content.

### Testing

1. Navigate to https://openlibrary.org/authors/OL59188A/Ayn_Rand and locate the canonical URL in the HTML
2. Verify canonical URL matches `<link rel="canonical" href="https://openlibrary.org/authors/OL59188A/Ayn_Rand">`
3. Click on one of the sort options on the page such as "Most Recent" and locate the canonical URL in the HTML
4. Verify canonical URL matches the one from step 2, and as such does not include `?sort=new#editions`

### Stakeholders
@jdlrobson @mekarpeles
